template/androidApp/src/main/java/io/bloco/template/features/details/DetailsViewModel.kt

```diff
@@ -1,11 +1,54 @@
 package io.bloco.template.features.details
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import io.bloco.core.domain.GetBook
+import io.bloco.core.domain.models.BookDetails
+import io.bloco.template.shared.PublishFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import javax.inject.Inject
 
 @HiltViewModel
 class DetailsViewModel @Inject constructor(
+    getBook: GetBook
+) : ViewModel() {
+    private val events = PublishFlow<Event>()
 
-): ViewModel() {
+    private val _bookDetails = MutableStateFlow(BookDetails(""))
```

Comment:
we can start using 1.7 here I think

think we can use internals here 
![imagem](https://user-images.githubusercontent.com/20781703/196188390-6d2b39e9-cc23-4a7e-bd34-f4d9b10b6399.png)

Reply:
It was a good idea, just tested it. 
Still experimental and needs K2 compiler, currently it doesn't seem to be compatible with the Android plugin